docs/v0.0.1/layout-specs.md

```diff
@@ -41,90 +82,182 @@ Each element in the hierarchical structure contains the following key-value pair
-
-![image](img/layout-nested-empty.png?raw=true)
-
-### Adding items to a nested element
+### Adding leaf items to a nested element
 The following example adds three new data types under `Watch` element:
 
 ```json
 {
-  "elements": [
+  "sections": [
     {
-      "text": "Display",
-      "iconUrl": "Category.Display.svg",
-      "elementType": "category",
-      "include": [
-        { "path": "DSCore.Color" },
-        { "path": "DSCore.ColorRange2D" }
-      ],
+      "text": "default",
+      "iconUrl": "",
+      "elementType": "section",
+      "showHeader": false,
+      "include": [],
       "childElements": [
         {
-          "text": "Watch",
-          "iconUrl": "Watch",
-          "elementType": "none",
+          "text": "Display",
+          "iconUrl": "/src/resources/icons/Category.Display.svg",
+          "elementType": "category",
           "include": [
-            { "path": "Core.View.Watch" },
-            { "path": "Core.View.Watch Image" },
-            { "path": "Core.View.Watch 3D" }            
+            {
+              "path": "DSCore.Color",
+              "iconUrl": "/src/resources/icons/DSCore.Color.png"
+            },
+            {
+              "path": "DSCore.ColorRange",
+              "iconUrl": "/src/resources/icons/DSCore.ColorRange.png"
+            }
+          ],
+          "childElements": [
+            {
+              "text": "Watch",
+              "iconUrl": "Watch",
+              "elementType": "group",
+              "include": [
+                { "path": "Core.View.Watch" },
+                { "path": "Core.View.Watch Image" },
+                { "path": "Core.View.Watch 3D" } 
+              ]
+            }   
           ]
-        }        
-      ]
+        }]
     }]
 }
 ```
 
 This results in library view that looks like the following image:
 
-![image](img/layout-nested-items.png?raw=true)
+![image](img/layout-leaf-items.png?raw=true)
+
+### The Miscellaneous section
+As mentioned, this layout specification works closely with `loadedTypes`. The `Miscellaneous` section, which displays left-over items, only shows up if `loadedTypes` contains items that are unspecified in this layout specification.
+
+The tree structure of this section is generated based on the `fullyQualifiedName` of the items. As an example, if there is a node called `Core.Web.Web Request` as left-over, the "Core" category is created:
+
+![image](img/layout-miscellaneous.png?raw=true)
+
+This is how it looks when the "Core" category is fully expanded:
+![image](img/layout-miscellaneous-expanded.png?raw=true)
+*Note: The general rule is that all items in `loadedTypes` should be specified so that `Miscellaneous` section will not appear in the library.*
+
+### Appending new sections
+New sections can be appended at the same level as `default` and `Miscellaneous` sections. As an example, we can add a new section called `Add-ons` that displays items with names that start with `pkg://`:
```

Comment:
Let's avoid *we* in this documentation, I'd propose the following tweaks:

*In addition to the `default` and `Miscellaneous` sections, custom sections can also be added alongside them. The following snippet shows how to add a new `Add-ons` section which displays all items with names that start with `pkg://`*